File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/simple_factory/DivOperation.java
package jv.com.tencent.cain.design_pattern.simple_factory;

/**
 * @author cainjiang
 * @date 2019-02-22
 * 除法操作类
 */
public class DivOperation extends Operation {
    @Override
    public double getResult() {
        if (numB == 0.0) {
            throw new IllegalArgumentException("被除数不能为0!");
        } else {
            return numA / numB;
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/simple_factoty/DivOperation.kt
package kt.com.tencent.cain.desigin_pattern.simple_factoty

/**
 * @author cainjiang
 * @date 2019-02-22
 * 除法操作类
 */
class DivOperation : Operation() {
    override fun getResult(): Double {
        if (numB == 0.0) {
            throw IllegalArgumentException("除数不能为0!")
        } else {
            return numA / numB
        }
    }
}